### PR TITLE
Add RequestContextUtil (prep for non-static SerializationManager)

### PR DESCRIPTION
**Pull request by ReubenBond**
Please review the latest commit only and of course avoid merging before the prior commits are merged (#2654 & #2665).

Since `RequestContext` has an implicit dependency on `SerializationManager`, changes will need to be made to support #467.

This PR injects a seemingly useless class, `RequestContextUtil` which will later hold an instance of non-static `SerializationManager` (as can be seen in #2592).

It also adds a client-side dependency on `Microsoft.Extensions.DependencyInjection`. This needs to happen sooner or later. DI is not exposed to consumers by this PR.

### CODE REVIEW

**Comment by ReubenBond** [comment]
The goal here is to break up #2592. If this strategy is alright, then I'll do the same for `MessageFactory` and continue rebasing #2592 as chunks are split out. It's a tedious process, but hopefully it's easier to review these individual pieces.

**Comment by jdom** [comment]
Hmmm, not too fond of the RequestContextUtil name. After I see what it does, I'll try to help with a better name

File: src/Orleans/Runtime/RequestContextUtil.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+using System.Collections.Generic;
+
+namespace Orleans.Runtime
+{
+    public class RequestContextUtil
```
Comment (sergeybykov) [inline src/Orleans/Runtime/RequestContextUtil.cs]: I'm trying to wrap my head around the purpose of `RequestContextUtil`. What is `Export()` supposed to do eventually? Deep-copy `RequestContext`?

**Comment by sergeybykov** [comment]
`RequestContextUtil` -> `RequestContextManager`?

File: src/Orleans/Runtime/RequestContextUtil.cs (hunk 1 above)
Comment (ReubenBond) [inline src/Orleans/Runtime/RequestContextUtil.cs]: @sergeybykov that's right, so it needs a reference to SerializationManager. Currently the class is just a stub

**Comment by ReubenBond** [comment]
Sergey's suggestion is good, imo

**Comment by ReubenBond** [comment]
I think I'll close this PR and just pass SerializationManager around where needed, instead.

**Pull request closed without merging**